File: app/src/main/java/org/macho/beforeandafter/preference/DeleteAllDialog.java
package org.macho.beforeandafter.preference;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;

import com.google.android.gms.ads.InterstitialAd;
import com.google.android.gms.ads.MobileAds;

import org.macho.beforeandafter.AdUtil;
import org.macho.beforeandafter.R;
import org.macho.beforeandafter.RecordDao;

import java.io.File;

/**
 * Created by yuukimatsushima on 2017/10/09.
 */

public class DeleteAllDialog extends DialogFragment {
    private InterstitialAd interstitialAd;

    public static DialogFragment newInstance(Activity activity) {
        return new DeleteAllDialog();
    }
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        MobileAds.initialize(getActivity(), getString(R.string.admob_app_id));

        interstitialAd = new InterstitialAd(getActivity());
        AdUtil.getInstance().loadInterstitialAd(interstitialAd, getContext());

        return new AlertDialog.Builder(getActivity()).setTitle(R.string.delete_all_title)
                .setMessage(R.string.delete_all_confirmation_message)
                .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        RecordDao.getInstance().deleteAll();
                        for (File file : new File("/data/data/org.macho.beforeandafter/files").listFiles()) {
                            file.delete();
                        }
                        AdUtil.show(interstitialAd);
                    }
                })
                .setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        AdUtil.show(interstitialAd);
                    }
                })
                .create();
    }
}

File: app/src/main/java/org/macho/beforeandafter/preference/DeleteAllDialog.kt
package org.macho.beforeandafter.preference

import android.app.Activity
import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import com.google.android.gms.ads.InterstitialAd
import org.macho.beforeandafter.shared.util.AdUtil
import org.macho.beforeandafter.R
import org.macho.beforeandafter.shared.data.record.RecordDao
import org.macho.beforeandafter.shared.data.record.RecordDaoImpl
import org.macho.beforeandafter.shared.util.showIfNeeded
import java.io.File

class DeleteAllDialog: androidx.fragment.app.DialogFragment() {
    companion object {
        fun newInstance(activity: Activity): androidx.fragment.app.DialogFragment {
            return DeleteAllDialog()
        }
    }

    private var interstitialAd: InterstitialAd? = null

    private val recordDao: RecordDao = RecordDaoImpl() // TODO: take from Dagger

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        AdUtil.initializeMobileAds(context!!)
        interstitialAd = AdUtil.instantiateAndLoadInterstitialAd(context!!)

        return AlertDialog.Builder(activity).setTitle(R.string.delete_all_title)
                .setMessage(R.string.delete_all_confirmation_message)
                .setPositiveButton(R.string.ok) { dialogInterface, i ->
                    recordDao.deleteAll()
                    val fileArray = File(context!!.filesDir.toString()).listFiles()
                    if (fileArray != null) {
                        for (file in fileArray) {
                            file.delete()
                        }
                    }
                    interstitialAd?.showIfNeeded(context!!)
                }
                .setNegativeButton(R.string.cancel) { dialogInterface, i ->
                    interstitialAd?.showIfNeeded(context!!)
                }
                .create()
    }
}